pkg/model/model_test.go

```diff
@@ -0,0 +1,327 @@
+package model_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"crypto/rand"
+	"encoding/hex"
+
+	diskfs "github.com/diskfs/go-diskfs"
+	losetup "github.com/freddierice/go-losetup/v2"
+	"github.com/metal-toolbox/vogelkop/pkg/model"
+)
+
+func cleanupTestDisk(image_name string, loopdev *losetup.Device) (err error) {
+	if err = loopdev.Detach(); err != nil {
+		return
+	}
+
+	if err = os.Remove(image_name); err != nil {
+		return
+	}
+
+	return
+}
+
+func prepareTestDisk(size int64) (loopdev losetup.Device, disk_img string, err error) {
+	disk_img = tempFileName("", "")
+
+	var (
+		disk_size int64 = size * 1024 * 1024
+	)
+
+	_, err = diskfs.Create(disk_img, disk_size, diskfs.Raw)
+	if err != nil {
+		return
+	}
+
+	loopdev, err = losetup.Attach(disk_img, 0, false)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func createPartitions(bd *model.BlockDevice, partitions []*model.Partition) (out string, err error) {
+	for _, p := range partitions {
+		p.BlockDevice = bd
+
+		out, err = p.Create()
+
+		if err != nil {
+			return
+		}
+	}
+
+	// Since we are using loopback devices for this test, we need to use kpartx
+	// to make the partition devices accessible
+	out, err = kpartxAdd(bd.File)
+	if err != nil {
+		return
+	}
+
+	for _, p := range partitions {
+		var partition_bd *model.BlockDevice
+
+		partition_bd, err = model.NewBlockDevice(p.GetLoopBlockDevice())
+
+		if err != nil {
+			return
+		}
+
+		p.BlockDevice = partition_bd
+	}
+
+	return
+}
+
+func kpartxAdd(device string) (k_out string, err error) {
+	k_out, err = model.CallCommand("kpartx", "-a", device)
+	return
+}
+
+func kpartxDel(device string) (k_out string, err error) {
+	k_out, err = model.CallCommand("kpartx", "-d", device)
+	return
+}
+
+func tempFileName(prefix, suffix string) string {
+	randBytes := make([]byte, 16)
+	_, _ = rand.Read(randBytes)
+	return filepath.Join(os.TempDir(), prefix+hex.EncodeToString(randBytes)+suffix)
+}
+
+func TestConfigureRaid(t *testing.T) {
```

Comment:
recommend: tests should include non-happy path test cases - for when an error is expected.